research/object_detection/meta_architectures/detr_lib.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Defines libraries for the Detection Transformer model in TF 2.0.
+
+Model paper: https://arxiv.org/abs/2005.12872
+Transformer model code source: https://github.com/tensorflow/tensor2tensor
+"""
+import tensorflow as tf
+from object_detection.utils import shape_utils
+from official.nlp.modeling import layers
```

Comment:
quick question: does the layers.MultiHeadAttention work for your needs?
Please let us know if you have any insights. We are moving MultiHeadAttention to keras

Reply:
It should work for my needs, yes. There's nothing happening here that's different from regular multiheaded attention. That would be great. For now we may just keep this around because we need to get some things pushed, but after that we can switch to the keras one.

Reply:
We will update to the kera's version when that becomes available.